### ISSUE TITLE
Accessing Individual Gradients

### ISSUE THREAD

**Issue by mayank0403** [tensorflow/federated#2700]
Hi,

I want to access individual gradients generated by each client for every round of training and save them to a file (as numpy arrays, for example). I need this to be able to inspect the gradients - for example, how different do malicious gradients look compared to benign ones, and so on. 

I found this issue from 3 years ago: https://github.com/tensorflow/federated/issues/756 that raises a similar concern. However, the solution suggested isn't supported in TFF nightly anymore (`tff.federated_collect`). I tried using `tff.aggregators.federated_sample` instead. It gives me a collection of gradients with server placement, but I don't know how to go forward. 

Is there a way to convert federated tensors of type `FederatedType(TensorType(tf.float32, [None, ...]), PlacementLiteral('server'), True)` to numpy arrays? Or another way to store them to file for later inspection? The solution in https://github.com/tensorflow/federated/issues/756 also mentions packing the collection result into `tf.Data.Dataset`, but that solution isn't clear. Are there any pointers that can help?

Thanks!

**Comment by jkr26** [comment]
Hi Mayank!

I personally tend to think that the appropriate way to do this in TFF is simply to *return the clients-placed values*. Meaning, if your TFF does something like:

```
@tff.federated_computation(...)
def do_something(val_at_server):
  val_at_clients = tff.federated_broadcast(val_at_server)
  result_at_clients = tff.federated_map(work, val_at_server)
  aggregated_result = tff.federated_sum(result_at_clients)
  return aggregated_result, result_at_clients
```

then invoking this computation in the TFF runtime will return you a list representing the computed values at the clients. 

Hope this helps!

**Comment by mayank0403** [comment]
Thanks! That worked for my case.
